### PR TITLE
Allow prod brokers to answer debug applications

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java [hunk 1]
```diff
@@ -23,14 +23,18 @@
...
+import android.util.ArraySet;
 
 import com.microsoft.identity.common.WarningType;
 import com.microsoft.identity.common.adal.internal.util.StringExtensions;
 import com.microsoft.identity.common.internal.logging.Logger;
 
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java:35]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java [hunk 2]
```diff
@@ -23,14 +23,18 @@
 package com.microsoft.identity.common.adal.internal;
 
 import android.os.Build;
+import android.util.ArraySet;
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java:26]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java [hunk 3]
```diff
@@ -23,14 +23,18 @@
...
 import com.microsoft.identity.common.WarningType;
 import com.microsoft.identity.common.adal.internal.util.StringExtensions;
 import com.microsoft.identity.common.internal.logging.Logger;
 
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java:37]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java [hunk 4]
```diff
@@ -28,6 +28,7 @@
 import com.microsoft.identity.common.adal.internal.util.StringExtensions;
 import com.microsoft.identity.common.internal.logging.Logger;
 
+import java.util.Arrays;
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java:31]:
```suggestion
```

**Review by Om83 — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 5]
```diff
@@ -987,6 +987,25 @@
...
+         */
+        public static final String COMPANY_PORTAL_PROD_APP_SIGNATURE = BuildConfig.COMPANY_PORTAL_PROD_APP_SIGNATURE;//"1L4Z9FJCgn5c0VLhyAxC5O9LdlE=";
+
+        /**
+         * Signature info for Intune Company portal app that installs authenticator
+         * component.
+         */
+        public static final String BROKER_HOST_APP_SIGNATURE = BuildConfig.COMPANY_PORTAL_PROD_APP_SIGNATURE ;//"1L4Z9FJCgn5c0VLhyAxC5O9LdlE=";
+
+
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java:1008]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 6]
```diff
@@ -987,6 +987,25 @@
          */
         public static final String COMPANY_PORTAL_APP_SIGNATURE = BuildConfig.COMPANY_PORTAL_APP_SIGNATURE;//"1L4Z9FJCgn5c0VLhyAxC5O9LdlE=";
 
+        /**
+         * Apk packagename that will install AD-Authenticator. It is used to
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java:991]:
```suggestion
         * Apk packagename that will install InTune Company Portal. It is used to
```

**Comment by shahzaibj** [comment]
Can we also do something to make release applications talk to debug brokers? We wouldn't want to ship with that...but if we can do this just for testing purposes (perhaps controlled by a boolean command line arg) then that would be great

**Comment by AdamBJohnsonx** [comment]
> Can we also do something to make release applications talk to debug brokers? We wouldn't want to ship with that...but if we can do this just for testing purposes (perhaps controlled by a boolean command line arg) then that would be great

We could, but what does it get us?

**Comment by shahzaibj** [comment]
> > Can we also do something to make release applications talk to debug brokers? We wouldn't want to ship with that...but if we can do this just for testing purposes (perhaps controlled by a boolean command line arg) then that would be great
> 
> We could, but what does it get us?

When we do release testing, we create RC builds for MSAL, ADAL and these are release builds i.e. they can only talk to prod brokers....it would be good if we can get these RC release builds to also talk to BrokerHost for testing purposes.

**Comment by shahzaibj** [comment]
> > > Can we also do something to make release applications talk to debug brokers? We wouldn't want to ship with that...but if we can do this just for testing purposes (perhaps controlled by a boolean command line arg) then that would be great
> > 
> > 
> > We could, but what does it get us?
> 
> When we do release testing, we create RC builds for MSAL, ADAL and these are release builds i.e. they can only talk to prod brokers....it would be good if we can get these RC release builds to also talk to BrokerHost for testing purposes.

The same would be true for snapshot builds of MSAL

**Pull request merged**